### PR TITLE
feat(ci): clean up validation

### PR DESCRIPTION
**Pull request by smartcontracts**
**Description**
Heavily cleans up the validation process by upgrading the validation
script to be less tightly coupled to CI.

### CODE REVIEW

File: src/generate.js [hunk 1]
```diff
@@ -1,241 +1,64 @@
...
-const factoryABI = require("./abi/factory.json");
 
 /**
  * Base URL where static assets are hosted.
  */
 const BASE_URL = "https://ethereum-optimism.github.io";
 
 /**
- * Call this to make CI require manual review.
+ * Generates a token list from the data in the data folder.
```
Comment (smartcontracts) [inline src/generate.js:14]: Cuts generation down to literally just generating -- you can validate via `validate` if you want that

File: bin/cli.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
+const { Command } = require("commander");
+
+const { generate } = require("../src/generate");
+const { validate } = require("../src/validate");
+
+const program = new Command()
+
+program
```
Comment (roninjin10) [inline bin/cli.js:8]: Nice!  I'm going to copy this for something else

File: bin/cli.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      process.exit(1);
+    }
+
+    const warns = results.filter(r => r.type === 'warning');
+    if (warns.length > 0) {
+      for (const warn of warns) {
+        console.log(`warning: ${warn.message}`);
+      }
+    }
+  })
```
Comment (roninjin10) [inline bin/cli.js:37]: Way cleaner

**Review by roninjin10 — APPROVED** [review]
nit: all these js files could be ts files but lgtm

**Pull request merged**